cdk/src/open-data-platform/lambda/write-demographic-data-handler.ts

```diff
@@ -0,0 +1,233 @@
+// asset-input/src/open-data-platform/lambda/write-demographic-data-handler.js
+import {APIGatewayEvent, APIGatewayProxyCallback, Context} from 'aws-lambda';
+
+const AWS = require('aws-sdk');
+const parse = require('csv-parser');
+const S3 = new AWS.S3();
+const rdsDataService = new AWS.RDSDataService();
+
+const GEO_ID = 'GEOID';
+const RACE_TOTAL = 'RaceTotal';
+const WHITE_POPULATION = 'Estimate!!Total:!!White alone';
+const BLACK_POPULATION = 'Estimate!!Total:!!Black or African American alone';
+
+/// Prepared SQL statements.
+const CREATE_DEMOGRAPHICS_TABLE =
+  'CREATE TABLE IF NOT EXISTS demographics( \n' +
+  '   census_geo_id varchar(255) NOT NULL, \n' +
+  '   total_population real, \n' +
+  '   black_percentage real,\n' +
+  '   white_percentage real, \n' +
+  '   PRIMARY KEY(census_geo_id) \n' +
+  ');';
+
+const DELETE_DEMOGRAPHICS_TABLE =
+  'DELETE FROM demographics WHERE census_geo_id IS NOT NULL';
+
+/// Reads the S3 CSV file and returns [DemographicsTableRow]s.
+const parseS3IntoDemographicsTableRow =
```

Comment:
Is the purpose of this function to get *all* of the rows? If so, that's a lot of data to return in a single variable and defeats the purpose of processing the file as a stream.

But that's also not necessarily a problem as long as know that the files aren't too big to store in lambda memory. I'd call that out explicitly in the comment (and rename to include `Rows` plural).

Reply:
Alternative suggestion: execute the `INSERT INTO` query for each chunk as its read. Instead of reading the entire file *then* executing the query on the whole set.

Reply:
This is a good point. I'm going to try this and the db parsing in another PR!

Reply:
Also for posterity: a chunk is just one row